<a href="https://colab.research.google.com/github/maryandy2002/maryandy2002/blob/main/MetododeNewton.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Método de Newton
Se realizara una aproximacion a la raiz de la funcion $f(x)=cos(x)-x$ utilizando el método de Newton con un $p_0=0$

En este bloque importamos las librerías que se van a utilizar,
 para crear una tabla al final para visualizar los resultados y
 para codificar la función que usaremos para aproximar la raíz

In [17]:
from tabulate import tabulate as tb
import math as mt

Ahora se definen las funciones $funcion$ y $dfuncion$ que guardan la funcion $f(x)$ y $f´(x)$ respectivamente

In [18]:
def funcion(x):
  return mt.cos(x)-x
def dfuncion(x):
  return (-1)*mt.sin(x)-1

Aqui se definen nuestras listas
$n$ para ir contando la cantidad de iteraciones,
$e$ para ir calculando el error aproximado
$p$ que es donde iremos guardando las aproximaciones a la raíz que vayamos obteniendo y
$f$ es donde realizaremos las iteraciones del método de Newton usando que

$p_{i+1}=p_i-\frac{f(p_i)}{f´(p_i)}$

In [29]:
n=[0]
e=[0]
p=[0]
f=[]

En este bloque es donde se llevan a cabo todas las iteraciones del método de Newton

In [30]:
while (True):
  f.append(p[-1]-(funcion(p[-1])/dfuncion(p[-1])))
  p.append(f[-1])
  e.append(abs(p[-1]-p[-2]))
  if (e[-1]<0.000001):
    break
  n.append(n[-1]+1)

Finalmente aqui utilizmamos la librería $tabulate$ para hacer una recopilación de los resultados obtenidos en el proceso

In [31]:
table=[n,p,f,e]
print(tb((zip(*table)),headers=["n","pi","p(i+1)","e"]))

  n        pi    p(i+1)            e
---  --------  --------  -----------
  0  0         1         0
  1  1         0.750364  1
  2  0.750364  0.739113  0.249636
  3  0.739113  0.739085  0.011251
  4  0.739085  0.739085  2.77575e-05


Así, podemos concluir que la raiz para la funcion $cos(x)-x$ es en $x=0.739085$ con un $\varepsilon=0.00002$